spec/models/reducer_spec.rb

```diff
@@ -220,36 +213,36 @@ def reduce_into(extracts, reductions=nil)
       extract2 = create :extract,
         extractor_key: 'bbb', subject_id: subject.id, workflow_id: workflow.id
 
-      extracts_double = instance_double(ActiveRecord::Relation)
-
-      subject_reduction_double = instance_double(SubjectReduction,
-        workflow_id: workflow.id,
-        subject_id: subject.id,
-        reducer_key: 'aaa',
-        extract_ids: [extract1.id],
-        extracts: extracts_double,
-        data: "foo"
+      subject_reduction = create(
+        :subject_reduction,
+        reducible: workflow,
+        subject: subject,
+        reducer_key: 'bbb',
+        extracts: [extract1],
+        data: 'foo'
       )
 
-      extract_fetcher = instance_double(ExtractFetcher, extracts: [extract1, extract2])
-      allow(extract_fetcher).to receive(:strategy=)
-      reduction_fetcher = instance_double(ReductionFetcher, retrieve: [subject_reduction_double])
-
-      running_reducer = create :reducer,
+      running_reducer = create(
+        :placeholder_reducer,
         key: 'aaa',
         type: 'Reducers::PlaceholderReducer',
         topic: Reducer.topics[:reduce_by_subject],
         reduction_mode: Reducer.reduction_modes[:running_reduction],
         reducible_id: workflow.id,
-        reducible_type: "Workflow"
+        reducible_type: 'Workflow'
+      )
 
-      allow(running_reducer).to receive(:get_reduction).and_return(subject_reduction_double)
-      allow(running_reducer).to receive(:reduce_into).and_return(subject_reduction_double)
-      allow(running_reducer).to receive(:associate_extracts)
-      allow(subject_reduction_double).to receive(:data=)
+      reduction_state_double = Reducer::ReductionState.new(
```

Comment:
I don't want to have to set this up for any spec that uses the `reduce_into` interface...thoughts appreciated.